examples/quote_ident/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2018 Benjamin Fry <benjaminfry@me.com>
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+extern crate pg_extend;
+extern crate pg_extern_attr;
+
+use pg_extend::guard_pg;
+use pg_extend::{pg_magic, pg_sys};
+use pg_extern_attr::pg_extern;
+
+// This tells Postgres this library is a Postgres extension
+pg_magic!(version: pg_sys::PG_VERSION_NUM);
+
+/// The pg_extern attribute wraps the function in the proper functions syntax for C extensions
+#[pg_extern]
+pub fn rust_quote_ident(value: String) -> String {
+    use std::ffi::{CStr, CString};
+
+    unsafe {
+        guard_pg(|| {
+            CStr::from_ptr(pg_sys::quote_identifier(
```

Comment:
technically this is the only thing that needs to be guarded, though no reason to force that.

Reply:
Who "owns" the memory here? PG? Do we need to free the CStr pointer?